### ISSUE TITLE
Azure Failures

### ISSUE THREAD

**Issue by mike-england** [Azure/azure-sdk-for-python#34616]
- **Package Name**: azure-monitor-opentelemetry
- **Package Version**: 1.3.0
- **Operating System**: Debian (bookworm)
- **Python Version**: 3.10

**Describe the bug**
I'm worried I'm opening up an old 'bug': https://github.com/Azure/azure-sdk-for-python/issues/33741 but I'm still having problems with Failures being reported to Azure Application Insights.  The opentelemtry-python-contrib project has updated to include support for Flask 3 (version 1.23.0/0.44b0) and I can confirm that it's working as expected.  However, when I include `OTEL_PYTHON_EXCLUDED_URLS` to keep one of my endpoints from reporting, they're still showing up as a request and a failed one

**To Reproduce**
I'm going to include some sample code in a docker file that you can try:

app.py:
```
from azure.monitor.opentelemetry import configure_azure_monitor

configure_azure_monitor()

import flask

app = flask.Flask(__name__)

@app.route("/")
def test():
    print('got one')
    return "Test flask request"

@app.route("/health")
def health():
    print('health endpoint')
    return "Health request"

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8080)
```

Dockerfile:
```
FROM python:3.10-slim

ENV PYTHONDONTWRITEBYTECODE=1
ENV PYTHONUNBUFFERED=1

COPY requirements.txt .
RUN python -m pip install -r requirements.txt

WORKDIR /app
COPY . /app

RUN adduser -u 5678 --disabled-password --gecos "" appuser && chown -R appuser /app
USER appuser

CMD ["python", "app.py"]
```

requirements.txt:
```
Flask==3.0.0
azure-monitor-opentelemetry==1.3.0
```

And to run it:
```
docker build -t azuretelemetry . && docker run --rm -p 8080:8080 -e APPLICATIONINSIGHTS_CONNECTION_STRING=$APPLICATIONINSIGHTS_CONNECTION_STRING -e OTEL_EXPERIMENTAL_RESOURCE_DETECTORS=$OTEL_EXPERIMENTAL_RESOURCE_DETECTORS -e OTEL_PYTHON_EXCLUDED_URLS=$OTEL_PYTHON_EXCLUDED_URLS azuretelemetry
```
Where `OTEL_PYTHON_EXCLUDED_URLS="health,swagger.*"`

**Expected behavior**
Endpoint requests are ignored

**Screenshots**
<img width="1438" alt="Screenshot 2024-03-04 at 4 01 37 PM" src="https://github.com/Azure/azure-sdk-for-python/assets/38864010/5d3126ff-3f6e-425f-98eb-62a385adc65d">

Any suggestions?  Is this a different project that would need to take care of this?  Am I doing something wrong?

**Comment by kashifkhan** [comment]
Thank you for the feedback @mike-england . We will investigate and get back to you asap.

**Comment by jeremydvoss** [comment]
I'll be looking into this.

**Comment by jeremydvoss** [comment]
I've confirmed that the OTEL_PYTHON_EXCLUDED_URLS works as intended with full URLs, URL endings, and even wildcards. This is a question for the Flask instrumentation itself. So, I recommend creating an issue in the [OTel contrib repo](https://github.com/open-telemetry/opentelemetry-python-contrib/issues) instead. But, let's see if we can solve it here. can you provide the full urls of the telemetry that you want to be filtered? could you also confirm that the environment variable is successfully being set in docker and visible from the app?

**Comment by mike-england** [comment]
Thanks for looking into this Jeremy.  I can confirm the variables being set:

# env
OTEL_PYTHON_EXCLUDED_URLS=health,swagger.*
OTEL_EXPERIMENTAL_RESOURCE_DETECTORS=azure_app_service

I'm accessing my localhost endpoint: http://localhost:8080/health

I can also confirm that the requests are not showing under the operation
name, they just show as these phantom "None" response codes.

Are you able to run the docker image and code I sent?  Maybe there's
something different about my azure app insights setup?

**Comment by annatisch** [comment]
Hi @jeremydvoss  - if I'm following this correctly, it looks like this isn't an issue with the Azure components specifically. Can this be closed or is there still work for us to do in relation to this issue?

**Comment by mike-england** [comment]
For the record, this still shows up a lot in our azure telemetry.  Has anyone tried the docker images I included to reproduce the issue?

**Comment by rads-1996** [comment]
@mike-england I tried to run the application using the docker image and commands you shared and I do not see the "None" response code for which you see some random telemetry - 

<img width="3111" height="835" alt="Image" src="https://github.com/user-attachments/assets/fce8b6a0-04bc-4a68-911a-8cb972077cad" />

Can you confirm if you are still facing the issue? I am using azure-monitor-opentelemetry==1.8.0 and flask==3.1.2

**Comment by mike-england** [comment]
I do still have the problem.  To be clear, the endpoint in this example (test) works fine, it reports fine.  The problem is in the /health endpoint, which we use in our kubernetes environment as a health check.  That is ignored and that reports a problem in the telemetry.

It's been a while, so I upgraded my test environment to use the following requirements.txt:
```
Flask==3.1.2
waitress==3.0.2
azure-monitor-opentelemetry==1.8.1
flask-restx==1.3.2
```

**Comment by rads-1996** [comment]
> I do still have the problem. To be clear, the endpoint in this example (test) works fine, it reports fine. The problem is in the /health endpoint, which we use in our kubernetes environment as a health check. That is ignored and that reports a problem in the telemetry.
> 
> It's been a while, so I upgraded my test environment to use the following requirements.txt:
> 
> ```
> Flask==3.1.2
> waitress==3.0.2
> azure-monitor-opentelemetry==1.8.1
> flask-restx==1.3.2
> ```

@mike-england Can you share the problem reported in the telemetry, as per your previous responses, you were seeing some telemetry with the response code as "None", are you still getting that?

**Comment by mike-england** [comment]
Yes, it's exactly the same. Here's a curl command which returns status 200:

`curl -s -o /dev/null -w "%{http_code}" localhost:8080/health`

And in azure telemetry, I get the same status 0 response code

<img width="1054" height="465" alt="Image" src="https://github.com/user-attachments/assets/c0fa1cfb-5e81-4b65-a84d-176b44aa37fb" />

**Comment by rads-1996** [comment]
@mike-england We have investigated this further and found there was a bug in the flask upstream instrumentation where the server request duration metrics were being captured for excluded urls. I have raised a PR to fix the bug, and you can track the progress of the issue in the `opentelemetry-python-contrib` repo - https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2352 using the issue previously opened by you. I am closing this issue here as the bug is in upstream opentelemetry.
